server.ts

```diff
@@ -19,8 +19,149 @@ const {
 
 const agent = await login();
 
-function getEmbedImages(post: AtoprotoAPI.AppBskyFeedDefs.PostView) {
-  return AppBskyEmbedImages.isView(post.embed) ? post.embed.images : [];
+function getPost(
+  post: AtoprotoAPI.AppBskyFeedDefs.PostView,
+  fullMedia = false,
+) {
+  const author = post.author;
+
+  const text = (AppBskyFeedPost.isRecord(post.record))
+    ? processText(post.record)
+    : "";
+
+  const isReply = (AppBskyFeedPost.isRecord(post.record) && post.record.reply)
+    ? true
+    : false;
+
+  const media = (
+      //if post with media
+      AppBskyEmbedImages.isView(post.embed)
+    )
+    ? post.embed.images
+    : (
+        //if post with media and Quoted post with media
+        AppBskyEmbedRecordWithMedia.isView(post.embed) &&
+        AppBskyEmbedImages.isView(post.embed.media)
+      )
+    ? post.embed.media.images
+    : [];
+
+  const mediastr = (media.length > 0)
+    ? tag(
+      "div",
+      ...media.map((image) => {
+        return tag(
+          "figure",
+          `<img src="${
+            fullMedia ? image.fullsize : image.thumb
+          }"/><figcaption>${image.alt}</figcaption>`,
```

Comment:
Elements with closing tags can use the `tag()` function, not only here but also in other parts.
To improve readability, I ask that you use `tag()` in your tags whenever possible, including where `<a>`, `<b>`, `<i>`, `<figre>` or `<figcaption>` tags are used.
```suggestion
          `<img src="${fullMedia ? image.fullsize : image.thumb}"/>`,
          tag("figcaption", image.alt),
```